### PR TITLE
[AIRFLOW-1018] Make processor use logging framework

### PR DESCRIPTION
**Pull request by bolkedebruin**
Dear Airflow maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [X] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1018

### Description
- [X] Here are some details about my PR, including screenshots of any UI changes:

Until now, the dag processor had its own logging implementation,
making it hard to adjust for certain use cases like working
in a container.

This patch moves everything to the standard logging framework.

### Tests
- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Tests added

### Commits
- [X] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

cc: @saguziel @criccomini @Fokko

### CODE REVIEW

File: airflow/jobs.py [hunk 1]
```diff
@@ -574,6 +547,10 @@ def __init__(
 
         self.heartrate = conf.getint('scheduler', 'SCHEDULER_HEARTBEAT_SEC')
         self.max_threads = conf.getint('scheduler', 'max_threads')
+
+        if log:
```
Comment (Fokko) [inline airflow/jobs.py:551]: Why is this required?

**Review by Fokko — COMMENTED** [review]
Very nice PR, have you tested this in prod?

File: airflow/jobs.py (hunk 1 above)
Comment (bolkedebruin) [reply airflow/jobs.py:551]: Because the log is only overridden when it is launched from the DagProcessor. We could move it into the Scheduler, but that would mean the handler would always kick in. Maybe not what we want.

**Comment by bolkedebruin** [comment]
@AllisonWang @saguziel The s3 tests for logging never ran and don't make sense. I am planning to disable them except if you have an idea how to fix them?

**Comment by allisonwang** [comment]
@bolkedebruin totally agree. I wasn't able to find a good way of testing S3 or GCS with unit tests. If the mock S3 test doesn't work we could just remove it.

**Comment by Fokko** [comment]
LGTM, very nice addition to cleaning up the logging
